components/expression/expression-renderer.js

```diff
@@ -254,19 +350,29 @@ genotet.ExpressionRenderer.prototype.drawGeneProfiles_ = function() {
  * @private
  */
 genotet.ExpressionRenderer.prototype.getHeatmapLabelSizes_ = function() {
-  // TODO(liana): Computes the horizontal/vertical text margin allocated for
-  // the heatmap labels.
-  if (!this.data.options.showGeneLabels) {
-    this.geneLabelWidth_ = 0;
-  } else {
-    // To implement... replace dummy value.
-    this.geneLabelWidth_ = 30;
+  var heatmapData = this.data.matrix;
+  var geneLabelsData = heatmapData.geneNames;
+  var conditionLabelsData = heatmapData.conditionNames;
+  this.geneLabelWidth_ = 0;
+  this.conditionLabelHeight_ = 0;
+
+  if (this.data.options.showGeneLabels) {
+    for (var i = 0; i < geneLabelsData.length; i++) {
+      if (geneLabelsData[i].length > this.geneLabelWidth_) {
+        this.geneLabelWidth_ = geneLabelsData[i].length;
+      }
+    }
+    this.geneLabelWidth_ *= this.geneLabelWidthFactor_;
+    this.geneLabelWidth_ += this.labelMargin_;
   }
-  if (!this.data.options.showConditionLabels) {
-    this.conditionLabelHeight_ = 0;
-  } else {
-    // To implement... replace dummy value.
-    this.conditionLabelHeight_ = 50;
+  if (this.data.options.showConditionLabels) {
+    for (var i = 0; i < conditionLabelsData.length; i++) {
+      if (conditionLabelsData[i].length > this.conditionLabelHeight_) {
+        this.conditionLabelHeight_ = conditionLabelsData[i].length;
+      }
+    }
+    this.conditionLabelHeight_ *= this.conditionLabelHeightFactor_;
+    this.conditionLabelHeight_ += this.labelMargin_;
```

Comment:
Do not add margin to label height. LabelHeight is defined as the height of the label text itself, not including the margins.